parsers/mainz.py

```diff
@@ -0,0 +1,177 @@
+    '9': 'mit Süßungsmitteln',
+    '10': 'enthält eine Phenylalaninquelle',
+    'S' : 'Schweinefleisch',
+    'G' : 'Geflügelfleisch',
+    'R' : 'Rindfleisch',
+    'Gl' : 'Gluten',
+    'We' : 'Weizen (inkl. Dinkel)',
+    'Ro' : 'Roggen',
+    'Ge' : 'Gerste',
+    'Haf': 'Hafer',
+    'Kr' : 'Krebstiere und Krebstiererzeugnisse',
+    'Ei' : 'Eier und Eiererzeugnisse',
+    'Fi' : 'Fisch und Fischerzeugnisse',
+    'En' : 'Erdnüsse und Erdnusserzeugnisse',
+    'So' : 'Soja und Sojaerzeugnisse',
+    'La' : 'Milch und Milcherzeugnisse',
+    'Sl' : 'Sellerie und Sellerieerzeugnisse',
+    'Sf' : 'Senf und Senferzeugnisse',
+    'Se' : 'Sesamsamen und Sesamsamenerzeugnisse',
+    'Sw' : 'Schwefeldioxid und Sulfite > 10mg/kg',
+    'Lu' : 'Lupine und Lupinerzeugnisse',
+    'Wt' : 'Weichtiere und Weichtiererzeugnisse',
+    'Nu' : 'Schalenfrüchte',
+    'Man': 'Mandel',
+    'Has': 'Haselnüsse',
+    'Wa' : 'Walnüsse',
+    'Ka' : 'Kaschunüsse',
+    'Pe' : 'Pecanüsse',
+    'Pa' : 'Paranüsse',
+    'Pi' : 'Pistatien',
+    'Mac': 'Macadamianüsse',
+    'icon:S.png' : 'Scheinefleisch',
+    'icon:R.png' : 'Rindfleisch',
+    'icon:Fi.png' : 'Fisch',
+    'icon:Gl.png' : 'Glutenfrei',
+    'icon:La.png' : 'Lactosefrei',
+    'icon:Vegan.png' : 'Vegan',
+    'icon:Veggi.png' : 'Vegetarisch'
+    
+}
+
+def build_meal_name(meal):
+  # There are the extras of the meal inside the meal name
+  # This will remove the extras and the unnecessary spaces
+  # Example: 6 gebackene Fischstäbchen (Gl,Fi,We) mit Reis und veganem Joghurt-Kräuter-Dip (3,Gl,So,Sf,Ge)
+  # Output: 6 gebackene Fischstäbchen mit Reis und veganem Joghurt-Kräuter-Dip
+	name = ' '.join(re.sub(r'\((\w+,?)+\)', '', str(meal)).split())
+  
+  # Shorten the meal name to 250 characters like the api specification: https://doc.openmensa.org/feed/v2/#name
+	if len(name) > 250:
+			name = name[:245] + '...' 
+	
+	return name
+	
+def build_meal_notes(meal):
+  meal_name = str(meal.find('div', class_="speiseplanname").string).strip()
+  images = meal.find_all('img')
+
+  # Use a set for easy elimination of duplicates
+  notes = set()
+      
+  # extracting the icons with spezial informations about the meal
+  # Example: <img src="/fileadmin/templates/images/speiseplan/Veggi.png"/>
+  for icon in images:
+    if "icon:"+path.basename(icon['src']) in extraLegend:
+      notes.add(extraLegend["icon:"+path.basename(icon['src'])])
+
+  for l in extract_legend_notes.findall(meal_name):
+    if l in extraLegend:
+      notes.add(extraLegend[l])
+
+  return list(notes)
+	
+def build_meal_price(meal):
+	meal_prices = {}
+	
+	prices = price_regex.findall(str(meal))
+  # The pricing for employee and others are the same!
+	meal_prices["student"] = prices[0].replace(',', '.')
+	meal_prices["employee"] = prices[1].replace(',', '.')
+	meal_prices["other"] = prices[1].replace(',', '.')
+	
+	return meal_prices
+	
+def parse_data(canteen, data):	
+	for v in data.find_all('div'):
+		if not v.has_attr('class'):
```

Comment:
This algorithm implicitely assumes that the `div`s appear in a certain order. In case this order is violated in the future, I think it might be nice to have a comment here that reminds of that. Something like
```
# We assume that the `div`s appear in a certain order and will associate each meal to the previously encountered date and category.
```

Reply:
Thats a create idea. I've didn't saw another way to get the meals from the page.